src/OrchardCore/OrchardCore.Abstractions/BackgroundTasks/HttpContextExtensions.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace OrchardCore.BackgroundTasks;
+
+internal static class HttpContextExtensions
+{
+    public static void SetBaseUrl(this HttpContext context, string baseUrl)
```

Comment:
Can we check if the `baseUrl` is null or empty?

Reply:
We use `Uri.TryCreate()`